onyo/lib/commands.py

```diff
@@ -379,59 +378,45 @@ def get(inventory: Inventory,
-    #       - This suggests a generic filter_assets method (for an Inventory)
-    #       - Gets filters, possibly arbitrary callables (see filter(callable, list) in get_assets_by_query)
-    #       - Check usecases for whether that can cover all the queries
-    results = get_assets_by_query(inventory.repo.asset_paths,
-                                  keys=set(keys),
-                                  paths=paths,
-                                  depth=depth,
-                                  filters=filters)
-    # TODO: Move this inside query. A returned asset (-dict) should be filled accordingly already.
-    #       See TODO for UNSET_VALUE definition.
-    results = fill_unset(results, keys, UNSET_VALUE)
-
-    # TODO: use `natsort` package.
+    keys = deduplicate(keys) or inventory.repo.get_required_asset_keys()
+    results = inventory.get_assets_by_query(paths=paths,
+                                            depth=depth,
+                                            filters=filters)
+    results = list(fill_unset(results, keys))
+    # convert paths for output
+    for r in results:
+        r['path'] = r['path'].relative_to(inventory.root)
+
     results = natural_sort(
-        assets=list(results),
-        keys=keys if sort_ascending or sort_descending else None,
+        assets=results,
+        keys=keys if sort_ascending or sort_descending else ['path'],
         reverse=True if sort_descending else False)
 
+    # for now, always include path to match previous behavior:
+    # TODO: Should we hardcode 'path' here?
+    if 'path' not in keys:
+        keys.append('path')
+    # filter output for `keys` only
+    results = [{k: v for k, v in r.items() if k in keys} for r in results]
+
     if machine_readable:
         sep = '\t'  # column separator
-        for asset, data in results:
+        for data in results:
             values = sep.join([str(value) for value in data.values()])
-            ui.print(f'{values}{sep}{asset.relative_to(Path.cwd())}')
-    else:
-        console = Console()
+            ui.print(f'{values}')
+    elif results:
         table = Table(
             box=box.HORIZONTALS, title='', show_header=True,
             header_style='bold')
-
         for key in keys:
             table.add_column(key, no_wrap=True)
+        for data in results:
+            values = [str(value) for value in data.values()]
+            table.add_row(*values)
 
-        table.add_column('path', no_wrap=True)
-
-        if results:
-            for asset, data in results:
-                values = [str(value) for value in data.values()]
-                table.add_row(*values, str(asset.relative_to(Path.cwd())))
-
-            console.print(table)
-        else:
-            console.print('No assets matching the filter(s) were found')
+        ui.rich_print(table)
+    else:
+        ui.rich_print('No assets matching the filter(s) were found')
```

Comment:
No need to change something, but doing this via a `rich_print()` seems unnecessary to me.

Reply:
Yes, `rich_print` should go entirely. As it's docstring states, it's just a refactoring helper to let `get` still do the same thing (and I wasn't sure, whether it looks the same as a regular print in this case).
However, that's ultimately about streamlining `UI` further, not about the get command or this line.